models/stg_apple_search_ads.yml

```diff
@@ -2,266 +2,322 @@ version: 2
+        description: Date for daily report aggregation.
+        tests:
+          - not_null
       - name: impressions
-        description: Number of times your ad appeared in App Store search results
+        description: Number of times your ad appeared in App Store search results.
       - name: spend
         description: The sum of costs associated with the number of impressions served. Spend is measured in the currency used in the campaign.
       - name: currency
-        description: This currency value should match the respective organization's currency value
+        description: This currency value should match the respective organization's currency value.
       - name: new_downloads
-        description: App downloads from new users who have never before installed the respective app
+        description: App downloads from new users who have never before installed the respective app.
       - name: redownloads
-        description: Occurs when a user downloads respective app, deletes it, and downloads the same app again following a tap on an ad on the App Store, or downloads the same app to an additional device
+        description: Occurs when a user downloads respective app, deletes it, and downloads the same app again following a tap on an ad on the App Store, or downloads the same app to an additional device.
       - name: taps
-        description: The number of times ad was tapped by users within the reporting time period
+        description: The number of times ad was tapped by users within the reporting time period.
 
+  - name: stg_apple_search_ads__ad_history
+    description: Each record represents a version of an ad in Apple Search Ads.
+    tests:
+      - dbt_utils.unique_combination_of_columns:
+          combination_of_columns:
+            - ad_id
+            - modified_at
+    columns:
+      - name: created_at
+        description: Timestamp of when ad was created.
+      - name: modified_at
+        description: Timestamp of when the ad object was last modified.
+        tests:
+          - not_null
+      - name: organization_id
+        description: Organization ID associated with the record.
+      - name: campaign_id
+        description: Campaign ID associated with the record.
+      - name: ad_group_id
+        description: Ad group ID associated with the record.
+      - name: ad_name
+        description: Ad name associated with the record.
+      - name: ad_id
+        description: Ad ID associated with the record.
+        tests:
+          - not_null
+      - name: ad_status
+        description: The current status of the ad, values include ENABLED and PAUSED.
+  
+  - name: stg_apple_search_ads__ad_report
+    description: Each record represents the daily performance of an ad in Apple Search Ads.
+    tests:
+      - dbt_utils.unique_combination_of_columns:
+          combination_of_columns:
+            - ad_id
+            - date_day
+            - _fivetran_synced
```

Comment:
Is there a reason we are also including the `_fivetran_synced` field with our combo of columns uniqueness test? I would assume the date day and ad id would suffice for what we are trying to achieve.

Reply:
Great question! So this is the table that could potentially set ad_id's = -1 (presumably based on our results from your test because of default ad settings). So you are correct in that `ad_id` and `date_day` should be sufficient if the default ads behavior didn't yield a -1 value for the `ad_id`. However, I felt like adding `_fivetran_synced` here will help reduce test failures due to this behavior. I will document this in the `DECISIONLOG` just in case anyone has the same question down the line!

Reply:
Actually, looking more closely at the data, I could utilize `ad_group_id` in place of `_fivetran_synced`. Also referring to this piece of [Apple Search Ads Docs ](https://searchads.apple.com/help/ads/0077-create-ad-variations) it looks like it should be ok at the `ad_id` and `ad_group_id` level.

Reply:
Sounds great, thanks for these notes and for the update. This change makes sense to me 😄